pyfr/libs/conduit.py

```diff
@@ -0,0 +1,116 @@
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         POINTER(c_double), c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         POINTER(c_longlong), c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p), # maybe this is copying, external might be better
+        (c_void_p, 'conduit_node_print', c_void_p),
+        (c_void_p, 'conduit_node_print_detailed', c_void_p),
+    ]
+
+
+class Conduit:
+    def __init__(self):
+        self.lib = ConduitWrappers()
+
+
+class ConduitNode:
+    def __init__(self, ptr=None):
+        cd = Conduit()
+        self.lib = cd.lib
+
+        if ptr is not None:
+            self.ptr = ptr
+        else:
+            self.ptr = c_void_p(self.lib.conduit_node_create(None))
+
+        self._set_ptr = {ConduitNode: self.lib.conduit_node_set_path_node,
+                         str: self.lib.conduit_node_set_path_char8_str,
+                         np.dtype('float64'): self.lib.conduit_node_set_path_float64_ptr,
+                         np.dtype('float32'): self.lib.conduit_node_set_path_float32_ptr,
+                         np.dtype('int64'): self.lib.conduit_node_set_path_int64_ptr,
+                        }
+
+    def __del__(self):
+        if hasattr(self, 'lib') and self.ptr is not None:
+            self.lib.conduit_node_destroy(self.ptr)
+        self.ptr = None
+
+    def __getitem__(self, key):
+        ptr = c_void_p(self.lib.conduit_node_fetch(self.ptr, key.encode()))
+        return ConduitNode(ptr)
+
+    def __setitem__(self, key, value):
+        if isinstance(value, str):
+            _ = self._set_ptr[str](self.ptr, key.encode(), value.encode())
+        elif isinstance(value, ConduitNode):
+            _ = self._set_ptr[ConduitNode](self.ptr, key.encode(), value.ptr)
+        elif isinstance(value, int):
+            _ = self.lib.conduit_node_set_path_int32(self.ptr, key.encode(),
+                                                     c_long(value))
+        elif isinstance(value, float):
+            _ = self.lib.conduit_node_set_path_float32(self.ptr, key.encode(),
+                                                       c_float(value))
+        elif isinstance(value, (np.ndarray, np.generic)):
+            value_c = np.ascontiguousarray(value, dtype = value.dtype)
+            t = value_c.dtype
+            l = c_longlong(len(value))
```

Comment:
No need for a cast.  There is also a size property which does what you want (len only gives the size of the first axis, size gives the total number of elements in the entire array).